code/parachain/frame/assets-transactor-router/src/lib.rs

```diff
@@ -493,28 +494,20 @@ pub mod pallet {
 			nonce: u64,
 			asset_info: AssetInfo<T::Balance>,
 		) -> Result<Self::LocalAssetId, DispatchError> {
-			// NOTE: Asset ID generation rational found here:
-			// https://github.com/PoisonPhang/composable-poison-fork/blob/INIT-13/rfcs/0013-redesign-assets-id-system.md#local-asset-id-generation
-			let bytes = protocol_id
-				.into_iter()
-				.chain(nonce.to_le_bytes())
-				.collect::<Vec<u8>>()
-				.try_into()
-				.expect("[u8; 8] + bytes(u64) = [u8; 16]");
-			let asset_id = Self::LocalAssetId::from(u128::from_le_bytes(bytes));
+			let asset_id = T::AssetsRegistry::generate_asset_id(protocol_id, nonce);
 
 			T::AssetsRegistry::register_asset(asset_id, None, asset_info)?;
 
 			Ok(asset_id)
 		}
 
 		fn create_foreign_asset(
-			foreign_asset_id: Self::ForeignAssetId,
+			protocol_id: [u8; 8],
+			nonce: u64,
 			asset_info: AssetInfo<T::Balance>,
+			foreign_asset_id: Self::ForeignAssetId,
 		) -> Result<Self::LocalAssetId, DispatchError> {
-			let asset_id = Self::LocalAssetId::from(u128::from_be_bytes(
-				sp_io::hashing::blake2_128(&foreign_asset_id.encode()),
-			));
+			let asset_id = T::AssetsRegistry::generate_asset_id(protocol_id, nonce);
```

Comment:
maybe we should restruct create_foreign_asset because this code looks like AssetsRegistry register_asset extrinsic, but here we dont check if a new location is already in storage. register_asset extrinsic does this check but register asset function in RemoteAssetRegistryMutate doesnt check. We need a way to create asset with a predefined id for migration though @benluelo @PoisonPhang

Reply:
You can use this interface, just split the id into two [u8; 8]

Reply:
> We need a way to create asset with a predefined id for migration though

This still exists. It's exposed via `RemoteAssetRegistryMutate::register_asset` @kkast

Reply:
59aac6316f6ac15c667d51cf1b214c5f33cc6aca